spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryWithIntroductionInterceptorTests.java

```diff
@@ -0,0 +1,175 @@
+        Assertions.assertDoesNotThrow(developer::code);
+    }
+
+    /**
+     * The target object implements the Teacher interface, and in this case, you want to use JDK for dynamic proxying
+     */
+    @Test
+    public void testDelegatePerTargetObjectIntroductionInterceptorWithInterface() {
+        Teacher teacher = () -> System.out.println("teach");
+        ProxyFactory pf = new ProxyFactory();
+        DelegatePerTargetObjectIntroductionInterceptor dii = new DelegatePerTargetObjectIntroductionInterceptor(DeveloperImpl.class, Developer.class);
+        pf.addAdvice(dii);
+        pf.addInterface(Teacher.class);
+        pf.setTarget(teacher);
+
+        Object proxy = pf.getProxy();
+        Assertions.assertTrue(AopUtils.isJdkDynamicProxy(proxy));
+        Assertions.assertTrue(proxy instanceof Teacher);
+        Assertions.assertTrue(proxy instanceof Developer);
+
+        Teacher teacher1 = (Teacher) proxy;
+        Assertions.assertDoesNotThrow(teacher1::teach);
+
+        Developer developer = (Developer) proxy;
+        Assertions.assertDoesNotThrow(developer::code);
+    }
+
+    /**
+     * The target object does not implement any interfaces, so it is necessary to use CGLIB for proxying
+     */
+    @Test
+    public void testProxyFactoryWithoutInterface() {
+        People people = new People();
+        ProxyFactory pf = new ProxyFactory();
+        pf.setTarget(people);
+        Object proxy = pf.getProxy();
+
+        Assertions.assertTrue(AopUtils.isCglibProxy(proxy));
+        Assertions.assertTrue(proxy instanceof People);
+        Assertions.assertDoesNotThrow(((People)proxy)::eat);
+
+        pf.addInterface(Teacher.class);
+        proxy = pf.getProxy();
+        Assertions.assertTrue(AopUtils.isCglibProxy(proxy));
+        Assertions.assertTrue(proxy instanceof Teacher);
+        Assertions.assertTrue(proxy instanceof People);
+        Assertions.assertDoesNotThrow(((People)proxy)::eat);
+    }
+
+    /**
+     * When the target object implements the Teacher interface
+     * but we have not explicitly called the addInterface method,
+     * we expect to use CGLIB; however, after calling it, we expect to use JDK
+     */
+    @Test
+    public void testProxyFactoryWithInterface() {
+        Teacher teacher = () -> System.out.println("teach");
+        ProxyFactory pf = new ProxyFactory();
+        pf.setTarget(teacher);
+        Object proxy = pf.getProxy();
+
+        Assertions.assertTrue(AopUtils.isCglibProxy(proxy));
+        Assertions.assertTrue(proxy instanceof Teacher);
+        Assertions.assertDoesNotThrow(((Teacher)proxy)::teach);
+
+        pf.addInterface(Teacher.class);
+        proxy = pf.getProxy();
+        Assertions.assertTrue(AopUtils.isJdkDynamicProxy(proxy));
+        Assertions.assertTrue(proxy instanceof Teacher);
+        Assertions.assertDoesNotThrow(((Teacher)proxy)::teach);
+    }
+
+    public static class People {
+        void eat() {
+            System.out.println("eat");
```

Comment:
We do not use `System.out.println()` in tests.

Please change the `void` methods to return the `String` instead.